server/src/main/java/org/opensearch/search/pipeline/SearchPipelineService.java

```diff
@@ -426,4 +428,12 @@ static class PipelineHolder {
             this.pipeline = Objects.requireNonNull(pipeline);
         }
     }
+
+    private boolean isFeatureEnabled() {
```

Comment:
@msfroh since feature flags are on JVM level (they are not dynamic), I think we could use just simple `enabled` property that will be passed to constructor (as `FeatureFlags.isEnabled(FeatureFlags.SEARCH_PIPELINE)` from core or manually during testuing), wdyt?